src/ucp/rma/basic_rma.c

```diff
@@ -423,20 +423,25 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_worker_fence, (worker), ucp_worker_h worker)
 UCS_PROFILE_FUNC(ucs_status_t, ucp_worker_flush, (worker), ucp_worker_h worker)
 {
     unsigned rsc_index;
+    unsigned flags = 0;
 
     UCP_THREAD_CS_ENTER_CONDITIONAL(&worker->mt_lock);
 
     while (worker->stub_pend_count > 0) {
         ucp_worker_progress(worker);
     }
 
+    if (worker->context->config.features & UCP_FEATURE_FAULT_TOLERANCE) {
+        flags |= UCT_FLUSH_FLAG_ACTIVE;
```

Comment:
this is not necessarily true - flush guarantees completion of previous RMA/AMO, not the current state of remote side. We can extend the UCP flush API with flags, or add another UCP call for "keepalive", for that purpose.